chainerx_cc/chainerx/routines/loss.cc

```diff
@@ -0,0 +1,49 @@
+#include "chainerx/routines/loss.h"
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "chainerx/array.h"
+#include "chainerx/backend.h"
+#include "chainerx/backprop_mode.h"
+#include "chainerx/backward_builder.h"
+#include "chainerx/backward_context.h"
+#include "chainerx/device.h"
+#include "chainerx/macro.h"
+#include "chainerx/routines/arithmetic.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/routines/explog.h"
+#include "chainerx/routines/logic.h"
+#include "chainerx/routines/misc.h"
+#include "chainerx/routines/reduction.h"
+#include "chainerx/scalar.h"
+
+namespace chainerx {
+
+Array SigmoidCrossEntropy(const Array& x1, const Array& x2, bool normalize) {
```

Comment:
I'm concerned with the absence of `'reduce'` argument which determines whether to reduce the shape of the input.

ref: https://docs.chainer.org/en/stable/reference/generated/chainer.functions.sigmoid_cross_entropy.html

Reply:
Yes @crcrpar . I do plan to add the implementation for `reduce` in this PR  Sorry I had forgotten to add a comment on the PR. I have added a TODO above.

Reply:
We thought that the reduction argument complicates the interface (e.g. in Chainer, different functions offer different reductions but it doesn't have to be that way) and choose not to introduce it when not needed in a discussion here https://github.com/chainer/chainer/pull/7063#issuecomment-496087996 for ChainerX.

Reply:
I think `normalize` can also be omitted from the function.

Reply:
If normalize is omitted, what should be the default behavior of the function? (Normalize=True or Normalize=False)

Reply:
You can simply omit the division, as well as `Sum` and `ignore_mask *`.
That is, you can just return this expression: `-(x1 * (x2 - (GreaterEqual(x2, ZerosLike(x2)))) - Log1p(Exp(-Fabs(x1))))`

Reply:
Then the behavior wouldn't be identical to either of `normalize=True` or `normalize=False`. So, tests cannot use `ChainerOpTest`. So, would `NumpyOpTest` be preferred?

Reply:
You could either use `F.sigmoid_cross_entropy` with `normalize=False` and multiply the result by `len(x)`, or use `op_utils.OpTest`.